src/AST/TypeExtensions.cs

```diff
@@ -207,13 +207,7 @@ public static Type SkipPointerRefs(this Type t)
         /// </summary>
         public static Type GetPointee(this Type t)
         {
-            var ptr = t as PointerType;
-            if (ptr != null)
-                return ptr.Pointee;
-            var memberPtr = t as MemberPointerType;
-            if (memberPtr != null)
-                return memberPtr.QualifiedPointee.Type;
-            return null;
+            return (t as PointerType)?.Pointee ?? (t as MemberPointerType)?.QualifiedPointee.Type;
```

Comment:
We haven't switched to C# 6 yet so do not use ?. I think you'd best revert this whole piece.